### PR TITLE
z/TPF Fixes during initial sniff testing.

### PR DESCRIPTION
**Pull request by jjohnst-us**
z/TPF fixes during initial sniff testing.  Fixes unresolved functions.
Fixes Memory Leak.

Signed-off-by: jjohnst-us <jjohnst@us.ibm.com>

Signed-off-by: jjohnst-us <jjohnst@us.ibm.com>

### CODE REVIEW

File: runtime/port/linuxs390/j9gs.c [hunk 1]
```diff
@@ -89,12 +90,20 @@ j9gs_initialize(struct J9PortLibrary *portLibrary, struct J9GSParameters *gsPara
 	Trc_PRT_gs_initialize_Exit(ret_code, gsParams->flags);
 
 	return IS_GS_INITIALIZED(gsParams) ? 1 : 0;
+
+#else /* !defined(J9ZTPF) */
```
Comment (DanHeidinga) [inline runtime/port/linuxs390/j9gs.c]: Is there a reason to have the #else section here?  Can't the ztpf code just exclude the main body and avoid duplicating the TRC and return?  Provided the gsParams is initialized before calling the function, this should be fine.

File: runtime/port/linuxs390/j9gs.c [hunk 2]
```diff
@@ -114,4 +123,9 @@ j9gs_deinitialize(struct J9PortLibrary *portLibrary, struct J9GSParameters *gsPa
 	Trc_PRT_gs_deinitialize_Exit();
 
 	return IS_GS_INITIALIZED(gsParams) ? 0 : 1;
+
+#else /* !defined(J9ZTPF) */
```
Comment (DanHeidinga) [inline runtime/port/linuxs390/j9gs.c]: Same question as above.  Can we limit the ztpf-specific code paths by changing what's #ifdef'd out

File: runtime/port/linuxs390/j9gs.c (hunk 1 above)
Comment (jjohnst-us) [reply runtime/port/linuxs390/j9gs.c]: Update made.

File: runtime/port/linuxs390/j9gs.c (hunk 2 above)
Comment (jjohnst-us) [reply runtime/port/linuxs390/j9gs.c]: Update made.

**Comment by DanHeidinga** [comment]
Jenkins test sanity

**Pull request merged**